### PR TITLE
RFC: (RE-7015) Add cross_compiled utility matcher methods

### PR DESCRIPTION
**Pull request by ScottGarman**
With all the cross-compiling we're now doing, these methods will help us
to refactor the puppet-agent component configs to improve readability.

### CODE REVIEW

**Comment by shermdog** [comment]
I personally rather have this as a flag in the platform definition itself vs. matching deeper somewhere in the code.  Seems more difficult to maintain this way.

**Comment by ScottGarman** [comment]
I'm inclined to agree. Rethinking this...

File: lib/vanagon/platform.rb [hunk 1]
```diff
@@ -239,5 +239,21 @@ def is_windows?
...
+      return !!@name.match(/^huaweios-|-s390x$|-sparc$/)
+    end
+
+    # Utility matcher to determine is the platform is a cross-compiled Linux variety.
+    # Many of the customizations needed to cross-compile for Linux are similar, so it's
+    # useful to group them together vs. other cross-compiled OSes.
+    #
+    # @return [true, false] true if it is a cross-compiled Linux variety, false otherwise
+    def is_cross_compiled_linux?
+      return !!@name.match(/^huaweios|^(sles-\d\d-s390x|el-\d-s390x)$/)
```
Comment (stahnma) [inline lib/vanagon/platform.rb]: Couldn't this return an && of is_linux? and is_cross_compiled?

**Comment by mckern** [comment]
I think handling this in flags and writing helper methods to enable it would be easier (long-term) than managing specific platforms like this.

**Comment by ScottGarman** [comment]
Closing this PR in favor of PR-368

**Pull request closed without merging**